consensus/src/dag/dag_store.rs

```diff
@@ -253,8 +254,32 @@ impl Dag {
         }
     }
 
-    pub fn bitmask(&self) -> Vec<Vec<bool>> {
-        // TODO: extract local bitvec
-        todo!();
+    pub fn lowest_incomplete_round(&self) -> Option<Round> {
+        for (round, round_nodes) in &self.nodes_by_round {
+            if round_nodes.iter().any(|node| node.is_none()) {
+                return Some(*round);
+            }
+        }
+        None
+    }
+
+    pub fn bitmask(&self, target_round: Round) -> DagSnapshotBitmask {
+        let lowest_round = match self.lowest_incomplete_round() {
+            Some(round) => round,
+            None => {
+                return DagSnapshotBitmask::new(self.highest_round() + 1, vec![vec![
```

Comment:
I think we still have a problem. What r is the latest round we locally have. but the target round is r+3?

Reply:
That is fine. The bitmask only needs to record what exists in the local dag, and everything in future and not in bitmask doesn't exist? `DagSnapshotBitmask::has()` takes care of this.

Reply:
Yes, I think you are right.
But then why do you return all false in self.highest_round() + 1? Cannot you just return an empty vector?